src/datahelper.py

```diff
@@ -1,200 +1,404 @@
+    def similar(self, csv_column_header: list, metadata_key: str) -> str:
+        """
+        Check the similarity of metadata and new postgresql-conform column headers and match them. Return the
+        postgresql-conform column name.
+        :param csv_column_header: list of csv column headers, after postgresql-conform correction
+        :param metadata_key: metadata column key from metadata file
+        :return: postgresql-conform column name
+        """
+        similarity_criteria = 0.8
+        sim_dict = {}
+        for csv_header in csv_column_header:
+            sim_value = SequenceMatcher(None, csv_header, metadata_key).ratio()
+            sim_dict[(csv_header, metadata_key)] = sim_value
+            if sim_value == 1:
+                break
+
+        if max(sim_dict.values()) >= similarity_criteria:
+            return max(sim_dict, key=sim_dict.get)[0]
+
+        raise ValueError(
+            f"Your metadata column name: {metadata_key} - has no similarity with the postgresql-conform "
+            f"corrected csv column headers {csv_column_header}\n"
+            f"Postgresql-conform corrected csv column headers cannot be inserted into metadata, due to missing "
+            f"match, please check manually if the column is present in your metadata.\n"
+            f"Similarity below {similarity_criteria}: {sim_dict}"
+        )
 
-    def make_each_string_col_unique_values(self):
+    def to_dataframe(self) -> object:
         """
-        This method reduces each column to annotate to its unique values.
-        :return: df
+        Return pd.DataFrame as generator object - use one pd.DataFrame at a the time.
+        :return:DataFrame: generator object
         """
-        pass
 
-    def to_csv(self, df_dict=None):
+        yield from self.df_dict.values()
+
+    def to_csv(self, df_dict=None) -> None:
         """
-        The method saves a dataframe as csv. The df is stored as value in a dict with corresponding df name as key.
-        :param df_dict:
-        :return:
+        Save a pd.DataFrame to csv.
+        :param df_dict: Key -> filename; Value -> pd.DataFrame.
+        :return None
         """
         df_dict[1].to_csv(
-            path_or_buf=f"{self.oeo_annotation_path}/{df_dict[0]}",
+            path_or_buf=f"{self.output_dir}/{df_dict[0]}.csv",
             index=False,
             encoding="utf-8",
             sep=";",
         )
 
-    ### metadata related-methods -> extra class
-
-    def read_annotation_pairs(self):
+    def read_metadata_json(self, path: str = None, debug: str = None) -> object:
         """
-        The method reads user-defined columns and corresponding annotation columns with annotations.
-        :return:
+        Read json file.
+        :param path: Path to json file
+        :param debug: If set to "json" print json
+        :return: json file
         """
-        pass
+        with open(path, "r", encoding="utf-8") as file:
+            if debug == "json":
```

Comment:
This looks like logging to me!1
You should consider using the logging module and set logging level (i.e logging.DEBUG) a beginning